src/backend/query_builder.rs

```diff
@@ -376,16 +376,23 @@ pub trait QueryBuilder: QuotedBuilder {
         sql: &mut SqlWriter,
         _collector: &mut dyn FnMut(Value),
     ) {
-        write!(
-            sql,
-            "{}",
-            match select_distinct {
-                SelectDistinct::All => "ALL",
-                SelectDistinct::Distinct => "DISTINCT",
-                SelectDistinct::DistinctRow => "DISTINCTROW",
+        match select_distinct {
+            SelectDistinct::All => write!(sql, "{}", "ALL").unwrap(),
+            SelectDistinct::Distinct => write!(sql, "{}", "DISTINCT").unwrap(),
+            SelectDistinct::DistinctRow => write!(sql, "{}", "DISTINCTROW").unwrap(),
+            #[cfg(feature = "backend-postgres")]
```

Comment:
Sorry if I have not documented it somewhere. As a rule-of-thumb, backend does not make use of feature guard. Instead, we should add the relevant feature in the `postgres` backend itself. This file is only for features common across all backends.